Missionframework/modules/28_captive/fnc/fn_captive_preInit.sqf

```diff
@@ -0,0 +1,102 @@
+/*
+    KPLIB_fnc_captive_preInit
+
+    File: fn_captive_preInit.sqf
+    Author: KP Liberation Dev Team - https://github.com/KillahPotatoes
+    Date: 2019-09-10
+    Last Update: 2019-09-26
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+    Public: No
+
+    Description:
+        The preInit function defines global variables, adds event handlers and set some vital settings which are used in this module.
+
+    Parameter(s):
+        NONE
+
+    Returns:
+        Module preInit finished [BOOL]
+*/
+
+if (isServer) then {
+    ["Module initializing...", "PRE] [CAPTIVE", true] call KPLIB_fnc_common_log;
+
+/*
+    ----- Module Initialization -----
+*/
+
+    // Process CBA Settings
+    [] call KPLIB_fnc_captive_settings;
+
+    // Units surrender on sector capture
+    ["KPLIB_sector_captured", {[_this select 0] call KPLIB_fnc_captive_checkSector;}] call CBA_fnc_addEventHandler;
+
+    // Unit animation on arrest
+    ["KPLIB_captive_arrested", {
+        _this select 0 playMoveNow "AmovPercMstpSsurWnonDnon_AmovPercMstpSnonWnonDnon";
+        _this select 0 playMove "AmovPercMstpSnonWnonDnon_EaseIn";
+    }] call CBA_fnc_addEventHandler;
+
+    // Check for handcuffed enemys
+    ["ace_captiveStatusChanged", {[_this select 0, _this select 1, _this select 2] call KPLIB_fnc_captive_setAceCaptive}] call CBA_fnc_addEventHandler;
+
+};
+
+// Add load captive EH
+["KPLIB_captive_load", {
+    params [
+        ["_unit", objNull, [objNull]],
+        ["_vehicle", objNull, [objNull]]
+    ];
+    // Move unit in vehicle cargo
+    _unit moveInCargo _vehicle;
+}] call CBA_fnc_addEventHandler;
+
+// Add unload captive EH
+["KPLIB_captive_unload", {
+    params [
+        ["_unit", objNull, [objNull]],
+        ["_vehicle", objNull, [objNull]]
+    ];
+    // Unload the unit
+    moveOut _unit;
+    _unit switchMove "AmovPercMstpSnonWnonDnon_EaseIn";
+}] call CBA_fnc_addEventHandler;
+
+// Player section
+if (hasInterface) then {
+
+    // Add EH for the captive unload action
+    ["KPLIB_captive_loaded", {
```

Comment:
We could throw these events on ACE load unload request events, that way people can always use these events. Currently they will be not thrown if ace is used:

Ref:
https://github.com/acemod/ACE3/blob/master/addons/captives/functions/fnc_doLoadCaptive.sqf#L45
https://github.com/acemod/ACE3/blob/master/addons/captives/functions/fnc_doUnloadCaptive.sqf#L21

Reply:
If we want to use the same event names, we'll get some trouble with the "vanilla" lib events, as "KPLIB_captive_unloaded" needs the unit and the vehicle to remove the action from the vehicle.
The ACE event just gives the unit, so an additional check would be needed to exit the "remove action EH". Alternative we could name the events "KPLIB_captive_aceLoaded" or similar.

Reply:
Ok, let's leave it as is for now.